cli/command/container/signals.go

```diff
@@ -15,10 +14,16 @@ import (
 //
 // The channel you pass in must already be setup to receive any signals you want to forward.
 func ForwardAllSignals(ctx context.Context, cli command.Cli, cid string, sigc <-chan os.Signal) {
-	var s os.Signal
+	var (
+		s  os.Signal
+		ok bool
+	)
 	for {
 		select {
-		case s = <-sigc:
+		case s, ok = <-sigc:
```

Comment:
Need some analysis on what closes this channel. The fix is correct anyway but need to make sure there isn't something more as well.

Reply:
pkg/signal.StopCatch closes the channel. StopCatch is called in a defer in attach.go, start.go, and run.go

Reply:
That is the only function that closes the channel.